src/components/LicensePicker/utils.js

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+import parse from 'spdx-expression-parse'
+import isNil from 'lodash/isNil'
+import unset from 'lodash/unset'
+import get from 'lodash/get'
+import set from 'lodash/set'
+const NOASSERTION = 'NOASSERTION'
+
+// Shared methods appliable to License Picker
+export default class LicensePickerUtils {
+  static parseLicense(license) {
+    return license ? parse(license) : {}
+  }
+
+  // Returns a license string based on the rules in input, following the specification of https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60
+  static stringify(obj) {
+    if (!obj) return null
+    if (obj.hasOwnProperty('noassertion')) return NOASSERTION
+    if (obj.license) return `${obj.license}${obj.plus ? '+' : ''}${obj.exception ? ` WITH ${obj.exception}` : ''}`
+    const left =
+      obj.left && obj.left.conjunction && obj.left.conjunction.toLowerCase() === 'or'
+        ? `(${this.stringify(obj.left)})`
+        : this.stringify(obj.left)
+    const right =
+      obj.right && obj.right.conjunction && obj.right.conjunction.toLowerCase() === 'or'
+        ? `(${this.stringify(obj.right)})`
+        : this.stringify(obj.right)
+    return left && `${left} ${!isNil(right) && obj.conjunction ? `${obj.conjunction.toUpperCase()} ${right}` : ''}`
+  }
+
+  /**
+   * Creates a new License Object in the specified path
+   * @param  {} obj original rules object
+   * @param  {} path where to create the new group
+   * @returns updated object rules
+   */
+  static createGroup(obj, path) {
+    if (path.length > 1 && obj.hasOwnProperty(path[0]) && obj[path[0]].hasOwnProperty(path[1])) {
+      obj[path[0]] = this.createGroup(obj[path[0]], path.slice(1))
+      return obj
+    }
+    if (path.length === 0)
+      return {
+        left: obj,
+        conjunction: 'and',
+        right: this.createGroupObject()
+      }
+    return {
+      ...obj,
+      [path[0]]: {
+        ...obj[path[0]],
+        right: this.createRuleObject(obj[path[0]].conjunction, obj[path[0]].right, this.createGroupObject())
+      }
+    }
+  }
+
+  /**
+   * Creates a new License Rules in the specified path
+   * @param  {} conjunction used conjuction to merge the new rules objects
+   * @param  {} obj current object rules
+   * @param  {} path where apply the change
+   * @returns updated object rules
+   */
+  static createRules(conjunction, obj, path) {
+    if (path.length > 1 && obj.hasOwnProperty(path[0]) && obj[path[0]].hasOwnProperty(path[1])) {
+      obj[path[0]] = this.createRules(conjunction, obj[path[0]], path.slice(1))
+      return obj
+    }
+    if (path.length === 0) return this.createRuleObject(conjunction, obj.left || obj, obj.right && obj.right)
```

Comment:
`obj.right && orbj.right` will just give a boolean. (end of line). should this be `obj.right || obj`? or just `obj.right`? or `!!obj.right`?